core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/DetermineJoinDistributionType.java

```diff
@@ -122,6 +124,61 @@ static double getSourceTablesSizeInBytes(PlanNode node, Lookup lookup, StatsProv
                 .sum();
     }
 
+    private static double getApproximateOutputSizeInBytes(PlanNode node, Context context)
+    {
+        return getApproximateOutputSizeInBytes(node, context.getLookup(), context.getStatsProvider(), context.getSymbolAllocator().getTypes());
+    }
+
+    @VisibleForTesting
+    static double getApproximateOutputSizeInBytes(PlanNode node, Lookup lookup, StatsProvider statsProvider, TypeProvider typeProvider)
+    {
+        if (hasExpandingNodes(node, lookup)) {
```

Comment:
Do we care about expanding codes for nodes below `FirstKnownOutputSize` node?

I think we could just inline this check into `getFirstKnownOutputSizeInBytes` and prevent recursion if current node is expanding